feat: Update test scripts use JSON for deployed libs (SC-2306)

lucas-manuel opened the pull request:
# Description

Test suite runs faster with this change, since we don't have to deploy libraries on every run.

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog
## Function Signature Changes

## Features 

## Events

satyamakgec committed:
Refactor: new repo architecture

deluca-mike committed:
feat: Metadata

satyamakgec commented on config.json:
```diff
@@ -0,0 +1,243 @@
+{
+  "language": "Solidity",
```
Can we update the readme about how to generate this file as the developer setup.

satyamakgec commented on test.sh:
```diff
@@ -1,29 +1,32 @@
-export DAPP_LINK_TEST_LIBRARIES=1
+export DAPP_LINK_TEST_LIBRARIES=0
 export DAPP_REMAPPINGS=$(cat remappings.txt)
+export DAPP_LIBRARIES=" contracts/libraries/loan/v1/LoanLib.sol:LoanLib:0x51A189ccD2eB5e1168DdcA7e59F7c8f39AA52232 contracts/libraries/pool/v1/PoolLib.sol:PoolLib:0x2c1C30fb8cC313Ef3cfd2E2bBf2da88AdD902C30 contracts/libraries/util/v1/Util.sol:Util:0x95f9676A34aF2675B63948dDba8F8c798741A52a"
```
we can re-use the build.sh here again to avoid confusion.

lucas-manuel replied:
I wanted to keep them separate so we can have a separate job in CI to run the build

satyamakgec committed:
distribute tests

Commit:
feat: update test script to use json, update submodule

feat: update CI to do one run

feat: update Ci config file

feat: update Ci config file

feat: use one job

fix: use path

feat: print dir structure

feat: print more info

feat: use DAPP_LIBRARIES

feat: add remappings

feat: run both jobs

fix: update tests to use percent accuracy instead of decimal

fix: update submodule

feat: run build to check cache rm

feat: add git status check to CI

feat: compare cache handling jobs

feat: log dapp.sol.json

feat: update builds to compare caching

feat: rm comparison job, add more parallelism to account for new files

deluca-mike commented on build.sh:
```diff
@@ -1,11 +1,14 @@
 export DAPP_SOLC_VERSION=0.6.11
 export DAPP_SRC="contracts"
-export DAPP_BUILD_OPTIMIZE=1
+export DAPP_BUILD_OPTIMIZE=0
```
If `optimize` is disabled, then I'm pretty sure `runs` does not matter, correct?

deluca-mike commented on contracts/test/TestUtil.sol:
```diff
@@ -585,6 +585,22 @@ contract TestUtil is DSTest {
+        uint256 denominator = val0 == 0 ? val1 : val0;
+        bool check = ((diff * RAY) / denominator) < percentage * RAY / 10_000;
+
+        if (!check){
+            emit log_named_uint("Error: approx a == b not satisfied, accuracy digits ", percentage);
+            emit log_named_uint("  Expected", val0);
+            emit log_named_uint("    Actual", val1);
+            fail();
+        }
```
```suggestion
        if (check) return;
        
        emit log_named_uint("Error: approx a == b not satisfied, accuracy digits ", percentage);
        emit log_named_uint("  Expected", val0);
        emit log_named_uint("    Actual", val1);
        fail();
```

deluca-mike commented on test.sh:
```diff
@@ -1,31 +1,34 @@
 export DAPP_SOLC_VERSION=0.6.11
 export DAPP_SRC="contracts"
-export DAPP_BUILD_OPTIMIZE=1
+export DAPP_BUILD_OPTIMIZE=0
```
Ditto on `runs` being ignored if `optimize` disabled.

Commit:
feat: update readme

Commit:
feat: update test util functions

Commit:
feat: try config files in CI

Commit:
feat: cat json

Commit:
feat: try multiple paths

Commit:
feat: try new path

Commit:
feat: print dir

Commit:
feat: print dir

Commit:
feat: update path

Commit:
feat: cat dapp build script

Commit:
feat: echo DAPP_STANDARD_JSON

Commit:
feat: update ci and build script to use libraries and remapings

Commit:
feat: use sudo

Commit:
feat: check dapp version

Commit:
Updated config.yml

Commit:
Updated config.yml

Commit:
Updated config.yml

satyamakgec committed:
change the docker image
